auto_nag/scripts/component.py

```diff
@@ -73,22 +84,31 @@ def get_bugs(self, date='today', bug_ids=[]):
         # Get the encoded component.
         indexes = probs.argmax(axis=-1)
         # Apply inverse transformation to get the component name from the encoded value.
-        components = self.model.clf._le.inverse_transform(indexes)
+        suggestions = self.model.clf._le.inverse_transform(indexes)
 
         results = {}
-        for bug, prob, index, component in zip(bugs, probs, indexes, components):
+        for bug, prob, index, suggestion in zip(bugs, probs, indexes, suggestions):
             # Skip product-only suggestions that are not useful.
-            if '::' not in component and bug['product'] == component:
+            if '::' not in suggestion and bug['product'] == suggestion:
                 continue
 
-            component = self.model.CONFLATED_COMPONENTS_MAPPING.get(component, component)
+            suggestion = self.model.CONFLATED_COMPONENTS_MAPPING.get(suggestion, suggestion)
+
+            assert '::' in suggestion
+            i = suggestion.index('::')
+            suggested_product = suggestion[:i]
+            suggested_component = suggestion[i + 2:]
+
+            # When moving bugs out of the 'General' component, we don't want to change the product (unless it is Firefox).
+            if bug['component'] == 'General' and bug['product'] != suggested_product and bug['product'] != 'Firefox':
```

Comment:
nit: bug['product'] not in {suggested_product, 'Firefox'}

Reply:
I was thinking about that, but I was not sure which one would be more readable in this particular case